code/components/gta-core-five/src/CrashFixes.InvalidPickupCreation.cpp

```diff
@@ -67,4 +68,55 @@ static HookFunction hookFunction([]
 		// CPedModelInfo::SetupPedBuoyancyInfo doesn't check that FragType isn't null.
 		origCPedModelInfo__SetupPedBuoyancyInfo = hook::trampoline(hook::get_call(hook::get_pattern("45 33 C9 4C 8B C0 48 8B D3 E8", 0x9)), &CPedModelInfo__SetupPedBuoyancyInfo);
 	}
+
+	// Not present on 1604 and we don't care about builds between 1604 and 2060
+	if (xbr::IsGameBuildOrGreater<2060>())
+	{
+		// Test for a valid weapon component info pointer before de-referencing it.
+		static struct : jitasm::Frontend
+		{
+			intptr_t retFail;
+
+			void Init(intptr_t location)
+			{
+				retFail = location + 14;
+			}
+
+			void InternalMain() override
+			{
+				test(rbx, rbx);				// if ( rbx )
+				jnz("success");				// {
+											//		goto success
+											// }
+				mov(rax, retFail);			//
+				jmp(rax);					// continue;
+											//
+				L("success");				// success:
+											//
+				// * original code			//
+				mov(rax, qword_ptr[rbx]);	//
+											// [run original code]
+				mov(rcx, rbx);				//			
+				// * original code END		//
+											//
+				ret();
```

Comment:
Shouldn't this jump to the address after the original code, instead of returning?

Reply:
🤦 yes it should, as i'm jumping to the patch